ws-discovery/networking.go

```diff
@@ -54,6 +54,10 @@ func GetAvailableDevicesAtSpecificEthernetInterface(interfaceName string) []onvi
 		for _, xaddr := range endpoints {
 			xaddr := strings.Split(strings.Split(xaddr.Text(), " ")[0], "/")[2]
 			fmt.Printf("Onvif WS-Discovery: Find %s \n", xaddr)
+			ref := doc.Root().FindElements("./Body/ProbeMatches/ProbeMatch/EndpointReference/Address")
+			uuidElements := strings.Split(ref[0].Text(), ":")
+			uuid := uuidElements[len(uuidElements)-1]
```

Comment:
if the len(ref) = 0 or len(uuidElements) = 0, this change will cause panic.
generate a random uuid if extracting EndpointReference is failed

Reply:
@cloudxxx8 I agree that this code will panic, however i don't think generating a random UUID would make sense in this situation. Wouldn't it be better to return nil or empty? If the camera device service sees that empty, it could choose to generate one, but it should be the consumer of the data that chooses, not the source (this library).

Reply:
> Wouldn't it be better to return nil or empty? If the camera device service sees that empty, it could choose to generate one

Not sure. For example, if the user bought ten TAPO C200 cameras, and they don't have the UUID, I prefer to print a warning message and skip these cameras, and the user needs to contact the manufacturer of TAPO C200 cameras for WS-Discovery fixing.

>  but it should be the consumer of the data that chooses, not the source (this library).

Got it, this library just parses the XAddrs and UUID.

Reply:
@weichou1229 Right. I was just saying the device service _could_ generate one, not that they _should_. This library shouldn't be making the decision for you.

IMO I think this library should return the empty UUID, and let the calling program decide how to proceed. In the case of the device-onvif service, I agree it should print a warning and refuse to register it with EdgeX. That seems like a valid approach to me, especially considering the `EndpointReference`  is required by the WS-Discovery Spec.

_However_  there is a chance that the `Address` may not be in `urn:uuid:` or `uuid:` format. Older versions of the spec indicate it can be any type of URI as long as it is unique and does not change.